InvenTree/plugin/builtin/suppliers/tme.py

```diff
@@ -0,0 +1,60 @@
+"""The TMEPlugin is meant to integrate the TME API into Inventree.
+
+This plugin can currently only match TME barcodes to supplier parts.
+"""
+
+import logging
+import re
+
+from django.utils.translation import gettext_lazy as _
+
+from plugin import InvenTreePlugin
+from plugin.mixins import SupplierBarcodeMixin
+
+logger = logging.getLogger('inventree')
+
+
+class TMEPlugin(SupplierBarcodeMixin, InvenTreePlugin):
+    """Plugin to integrate the TME API into Inventree."""
+
+    NAME = "TMEPlugin"
+    TITLE = _("Supplier Integration - TME")
+    DESCRIPTION = _("Provides support for scanning TME barcodes")
+    VERSION = "1.0.0"
+    AUTHOR = _("InvenTree contributors")
+
+    def parse_supplier_barcode_data(self, barcode_data):
+        """Get supplier_part and barcode_fields from TME QR-Code or DataMatrix-Code."""
+
+        if not isinstance(barcode_data, str):
```

Comment:
There is a fair amount of repeated code in this `parse_supplier_barcode_data` method which could be pushed to the `SupplierBarcodeMixin` class